be/src/storage/vectorized/async_delta_writer_executor.h

```diff
@@ -0,0 +1,39 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include <bthread/execution_queue.h>
+
+#include "common/config.h"
+#include "util/threadpool.h"
+
+namespace starrocks::vectorized {
+
+// Used to run bthread::ExecutionQueue task in pthread instead of bthread.
+// Reference: https://github.com/apache/incubator-brpc/blob/master/docs/cn/execution_queue.md
+class AsyncDeltaWriterExecutor : public bthread::Executor {
+public:
+    static AsyncDeltaWriterExecutor* Instance() {
+        static AsyncDeltaWriterExecutor instance;
+        return &instance;
+    }
+
+    int submit(void* (*fn)(void*), void* args) override {
+        auto st = _thread_pool->submit_func([=]() { fn(args); });
+        LOG_IF(WARNING, !st.ok()) << st;
+        return st.ok() ? 0 : -1;
+    }
+
+private:
+    AsyncDeltaWriterExecutor() {
+        CHECK(ThreadPoolBuilder("AsyncDeltaWriterExecutor")
```

Comment:
CHECK will cause progress crash. I don't think it is a good way.

Reply:
> CHECK will cause progress crash. I don't think it is a good way.

This is an unrecoverable error, there is nothing we can do but quit

Reply:
> > CHECK will cause progress crash. I don't think it is a good way.
> 
> This is an unrecoverable error, there is nothing we can do but quit

I modified the code to return an error if the thread pool building failed, to avoid process crash, but all loading jobs will fail if this really happened.